### PR TITLE
LTP: fix test case sigrelse01 issue

### PR DESCRIPTION
**Pull request by shaikshavali1**
This test case creates a child process using "FORK_OR_VFORK".
Fork behavior is not supported in sgx-lkl.
Hence, the below modifications are performed.
	1. Removed the code related to child process creation.
	2. Moved child and parent-related code/functionality to a single function "verify_sigrelse"
	3. Signal with ID 34 is skipped from testing. This signal ID is used for internal signals in the thread library of sgx-lkl-musl.
	4. Removed pipe handling code which is used for interprocess communication between parent and child.
	5. Removed the code related to the timer, we already have 60sec top-level timer for the test.

### CODE REVIEW

File: tests/ltp/patches/fix_sigrelse_sigrelse01.patch [hunk 1]
```diff
@@ -0,0 +1,673 @@
...
+-static void getout(void);
+ int choose_sig(int sig);
+ 
+ #define TRUE  1
+@@ -141,29 +135,12 @@ int choose_sig(int sig);
+ #ifndef DEBUG
+ #define DEBUG 0
+ #endif
+-
+-#define CHILD_EXIT(VAL) ((VAL >> 8) & 0377)	/* exit value of child process */
```
Comment (davidchisnall) [inline tests/ltp/patches/fix_sigrelse_sigrelse01.patch]: I think this is now dead code, please leave it in: eventually we will want to revert these and it makes it much easier to understand the diffs if they only include functional changes.

File: tests/ltp/patches/fix_sigrelse_sigrelse01.patch [hunk 2]
```diff
@@ -0,0 +1,673 @@
...
+-#define SIG_CAUGHT 8
+-#define WRITE_BROK 16
+-#define HANDLE_ERR 32
+-
+ int TST_TOTAL = 1;		/* number of test items */
+ 
+ char *TCID = "sigrelse01";	/* test case identifier */
+-static char mesg[MAXMESG];	/* message buffer for tst_res */
+ static int pid;			/* process id of child */
+-static int pipe_fd[2];		/* file descriptors for pipe parent read */
```
Comment (davidchisnall) [inline tests/ltp/patches/fix_sigrelse_sigrelse01.patch]: We ought to be able to use pipes between threads, why was this change made?

File: tests/ltp/patches/fix_sigrelse_sigrelse01.patch [hunk 3]
```diff
@@ -0,0 +1,673 @@
...
++}
++
+ /***********************************************************************
+  *   M A I N
+  ***********************************************************************/
+@@ -192,9 +216,6 @@ int main(int argc, char **argv)
+ 	 * parse standard options
+ 	 */
+ 	tst_parse_opts(argc, argv, NULL, NULL);
+-#ifdef UCLINUX
```
Comment (davidchisnall) [inline tests/ltp/patches/fix_sigrelse_sigrelse01.patch]: Why is this removed?  We should be using the UCLINUX code paths because we are a uCLinux implementation.

**Review by davidchisnall — CHANGES_REQUESTED** [review]
This is difficult to review with extraneous changes in the diff, please remove them.

If we are not using the uCLinux changes, please document why and track it (is it because exec is the problem)?

File: tests/ltp/patches/fix_sigrelse_sigrelse01.patch [hunk 4]
```diff
@@ -0,0 +1,673 @@
+This test case creates a child process using "FORK_OR_VFORK".
```
Comment (hukoyu) [inline tests/ltp/patches/fix_sigrelse_sigrelse01.patch]: Please change patch file name to sigrelse01.patch 
You can run command
git mv fix_sigrelse_sigrelse01.patch sigrelse01.patch

**Review by hukoyu — CHANGES_REQUESTED** [review]
@shaikshavali1  please change the patch file name as requested. Please also address David's feedback.

File: tests/ltp/patches/fix_sigrelse_sigrelse01.patch (hunk 4 above)
Comment (shaikshavali1) [reply tests/ltp/patches/fix_sigrelse_sigrelse01.patch]: @hukoyu, I renamed the file to sigrelse01.patch.

**Comment by shaikshavali1** [comment]
@hukoyu, I addressed your review comment.
@davidchisnall, This patch is not using threads. Hence, the pipe related code is removed. uClinux code is removed because exec is the problem and I will update the same in the commit message.

**Comment by davidchisnall** [comment]
Thank you.  Please track this patch as one that can be reverted when #598 is fixed.

File: tests/ltp/patches/fix_sigrelse_sigrelse01.patch (hunk 1 above)
Comment (shaikshavali1) [reply tests/ltp/patches/fix_sigrelse_sigrelse01.patch]: This patch file will be removed/reverted once #598 is fixed.

File: tests/ltp/patches/fix_sigrelse_sigrelse01.patch (hunk 2 above)
Comment (shaikshavali1) [reply tests/ltp/patches/fix_sigrelse_sigrelse01.patch]: This patch is not using threads. Hence, the pipe related code is removed.

File: tests/ltp/patches/fix_sigrelse_sigrelse01.patch (hunk 3 above)
Comment (shaikshavali1) [reply tests/ltp/patches/fix_sigrelse_sigrelse01.patch]: uClinux code is removed because exec is the problem and I will update the same in the patch description.

Comment (shaikshavali1) [reply tests/ltp/patches/fix_sigrelse_sigrelse01.patch]: I updated the patch description.

**Comment by hukoyu** [comment]
@davidchisnall  if looks good can you approve so that I can continue to process this change. Thanks

**Comment by hukoyu** [comment]
We are making test changes in the https://github.com/lsds/ltp/sgx-lkl now.
This PR resubmitted here: https://github.com/lsds/ltp/pull/2
